sensible-proxy.go

```diff
@@ -9,19 +9,59 @@ package main
 import (
 	"bufio"
 	"container/list"
+	"fmt"
 	"io"
 	"log"
 	"net"
 	"os"
 	"os/signal"
 	"strings"
 	"syscall"
+	"time"
 )
 
 var appLog *log.Logger
 
+type LogData struct {
```

Comment:
nitpicky comment: Can we chuck this further down the file (below main) or in another file?